### DISCUSSION TITLE
Filter by relation not working.

### DISCUSSION THREAD

**Discussion by kelcheone** [pocketbase/pocketbase#2636] [Q&A]
I'm trying to fetch a list of items by a relation (userId) but it's not working. It returns an empty array yet there exists data items with the same user Id.
```typescript
const userId = pb.authStore.model?.id;
if (!userId) return;
const res = await pb.collection("projects").getFullList<Project>({
            filter: `userId="${userId}"`,
        })
```

This returns an empty array but removing the filter or using other nonrelational columns returns the data properly.
This is the image of the records:

![image](https://github.com/pocketbase/pocketbase/assets/80192475/805541be-763c-4155-a8ea-3f0b8207c4c2)

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it locally with v0.16.4 (_assuming that `userId` is a single `relation` field pointing to the default `users` auth collection_).

What PocketBase version do you use?
Does it work in your case if you apply the filter in the Admin UI search bar?
If it works in the Admin UI, then make sure that `pb.authStore.model?.id` returns the id you are expecting.

If that doesn't help, please paste your collections configuration as JSON (_you can export it from the Admin UI > Settings > Export collections_) and I'll try to investigate it further.

**Reply by kelcheone** [reply]
Thanks for the quick reply, I finally found out why it wasn't returning. This was because I had set the relation to multiple instead of single, after fixing this I was able to make the query correctly.
